packages/loader/container-loader/src/container.ts

```diff
@@ -748,6 +750,17 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         this.emit("warning", warning);
     }
 
+    private async raceToSummarize(): Promise<void> {
+        try {
+            await this.context.raceToSummarize();
+            this.context.dispose();
```

Comment:
I'd personally move context disposition code to caller of raceToSummarize(), and leave this function only about this.context.raceToSummarize(), mostly because quorum handler already deals with context reloading.
Or maybe vise version - move if /else logic form handler into here.

Reply:
This is the part I'm unsure about. @anthony-murphy help here too. And a few other points too.

I wasn't sure where to put this little piece of logic: Container (loader) vs. ContainerRuntime (runtime).

How to correctly parameterize host preference for refresh vs. hot swap reload.

Do we need to do anything special to indicate to host that they should refresh the page or is "contextDisposed" event sufficient?

Reply:
Do we have confidence that context reload will work for any scenario?
I.e. it will never work for unknown code, as you can't trust it to do clean cleanup.
I just wonder if we need that complexity here in our layers, and expose it to users as well.

Do we have a scenario in mind?
I can see it working for data-only scenarios, where code can be clearly recycled.
Is this enough for added complexity?

Reply:
we need context reload for nullRuntime back compat. i think we could consider making it opt in for other cases, like the host should explictly reload on contextDispose

Reply:
i don't think we can remove it. that would break back-compat, but we can hide it

Reply:
Back compat is a concern, but the only thing that is affected is file without summary. I think we can make the case that we do not care that much

Reply:
i disgree. i think we can internalize it, but i don't think we can get rid of it entirely

Reply:
What's the scenario where we need it? Is there anything other than corrupted file with 3 ops in storage that nobody opens?

Reply:
replay, and our snapshot tests